### ISSUE TITLE
Supplementary Views Rendering on Wrong Days

### ISSUE THREAD

**Issue by jonpjones** [CVCalendar/CVCalendar#324]
- [x] I have verified there are no duplicate active or recent bugs, questions, or requests
- [X] I have verified that I am using the latest version of CVCalendar.
- [X] I have given my issue a non-generic title.
## Expected Behavior

Every day prior to the current one should have a blue circular supplementary view, and all views past today's date should not have any view at all. 
## Current Behavior

The days that actually have the supplementary view applied are wrong. When trying to apply the supplementary views to days previous to the current one, this is the result that I'm getting:
![image](https://cloud.githubusercontent.com/assets/15958579/17591348/5e497610-5fa3-11e6-8583-93e0ac331ece.png)

Also, when I scroll between months, it crashes on a nil dayView value. This does not happen sans supplementary view. 

Sidebar: I have also noticed that when I apply the supplementary view to -all- dayViews, I still get weird behavior. This is especially noticeable when I scroll from one month to another. Example with supplementary view applied to all dayviews: 
![image](https://cloud.githubusercontent.com/assets/15958579/17591483/f6da1074-5fa3-11e6-9afe-e25c7151923f.png)

Next Month: 
![image](https://cloud.githubusercontent.com/assets/15958579/17591494/04f11e5a-5fa4-11e6-8b76-e5bfa89ba806.png)
## Possible Solution

Hard to say. Obviously, the supplementary views are being applied to the day views incorrectly. When trying to apply the sup view to past days, certain days of the week are always given a supplementary view no matter what.  The columns that are incorrectly given a supplementary view are consistent across simulators. 
## Steps to Reproduce (for bugs)
1.  Using pod version 1.2.9. Besides the supplementary views, the calendar is working as intended. My basic setup uses the .Month presentation mode and the first weekdays is set to .Sunday.
2.  I have set up the supplementary view delegate methods as follows: 

``` swift
    func supplementaryView(viewOnDayView dayView: DayView) -> UIView {
        let view = UIView()
        let sideLength = min(dayView.frame.width - 10,dayView.frame.height - 10)
        view.frame.size = CGSizeMake(sideLength, sideLength)
        view.center = dayView.center
        view.layer.cornerRadius = sideLength / 2
        view.backgroundColor = .lightBlue()
        view.clipsToBounds = true
        return view
    }

    func supplementaryView(shouldDisplayOnDayView dayView: DayView) -> Bool {
        if dayView.date.convertedDate() < NSDate() {
            return true
        }
        return false
    }
```

Note: I added an NSDate extension to make it comparable, but this is not affecting the bug in any way. 
## Context

I am trying to add a supplementary view to only particular dayViews. Namely, I would like all days prior to the current date to have a small blue circle around it (much like the circleview used for dots and selections, but with a different color).
## Your Environment
- Version used (Cocoapods spec): 1.2.9
- Device used:: Reproduced bug on 5, 6, 6s simulators
- Physical device (y/n):  No
- Link to your project (if available): Private Repository

**Comment by jonpjones** [comment]
Here's a small update: 

It looks like the application of supplementary views on weeks / months is off. 
I tried adding a view to only a specific month, like so:
```swift
 func supplementaryView(shouldDisplayOnDayView dayView: DayView) -> Bool {
        let date = CVDate(date: NSDate())
        if dayView.date != nil {
            if dayView.date.month == date.month {
                return true
            }
        }
        return false
    }
```
However, when run, this is the current month:
![image](https://cloud.githubusercontent.com/assets/15958579/17597670/840e4e66-5fbc-11e6-9df9-34c5d03fb5c0.png)

And the 'next month' is:
![image](https://cloud.githubusercontent.com/assets/15958579/17597683/9a5207ee-5fbc-11e6-88c3-0375cfcbf8c1.png)

For some reason, the supplementary views are being applied in an 'every-other' type fashion, since days that should have a supView on the current month instead have that view applied to the next month.

**Comment by nunesdennis** [comment]
I had the same problem and I tried to use the fix made by "gregazevedo", changing the "    insertSubview(preView, atIndex: 0)" for the "weekView.insertSubview(preView, atIndex: 0)", and it worked.

**Comment by elsesiy** [comment]
@nunesdennis Would you be so kind to submit a PR? This looks like a serious issue to me, so we should fix it asap.

**Comment by nunesdennis** [comment]
I am not sure if it works in every situation, but I just got the fix from here "sup view frame bug fix #247" , made by @gregazevedo, and I didn't have time to test it very well (I would recommend @gregazevedo to retry the PR in case there is anything else on his fix)
